lib/echo/echo_client.rb

```diff
@@ -94,6 +94,20 @@ def create_order(granule_query, option_id, option_name, option_model, user_id, t
       #return {order_id: 1234, count: 2000}
       catalog_response = cmr_client.get_granules(granule_query, token)
       granules = catalog_response.body['feed']['entry']
+      order_info = get_order_information(granules.map {|g| g['id']}, token).body
+
+      # drop granules from the order if the selected order option is not one of the supported options by this granule
+      dropped_granules = []
+      Array.wrap(order_info).each do |info|
+        info = info['order_information']
+        unless Array.wrap(info['option_definition_refs']).any? {|ref| ref['name'] == option_name}
+          granule = granules.select {|g| g['id'] == info['catalog_item_ref']['id']}
```

Comment:
This and line 106 make the algorithm O(n^2). 4 million iterations for an order size of 2000.